src/Symfony/Component/HttpKernel/HttpCache/Esi.php

```diff
@@ -162,7 +162,7 @@ public function process(Request $request, Response $response)
 
         // we don't use a proper XML parser here as we can have ESI tags in a plain text response
         $content = $response->getContent();
-        $content = preg_replace('#<esi\:remove>.*?</esi\:remove>#', '', $content);
+        $content = preg_replace('#<esi\:remove>[^<]*</esi\:remove>#', '', $content);
```

Comment:
Even better the original regex doesn't even work :flushed:
http://3v4l.org/M7eiZ